### DISCUSSION TITLE
Is it possible to make join between Views?

### DISCUSSION THREAD

**Discussion by OlegShNayax** [simolus3/drift#1867] [Q&A]
Hi,

I need to create view class that based on another views. 
Is it possible to make join between those views ? 

Thanks

It would be nice to have option to build views classes based on another views like this:
```

abstract class DriftMachineDetailsView extends View  {

  DriftMachineView get machineView;
  DriftDetailsView get machineDetailsView;


  Expression<int?> get providerId => machineView.providerId;
  Expression<String?> get languageId => machineDetailsView.details;


  @override
  Query as() => select([
    providerId,
    languageId
  ]).from(machineView).join([
        innerJoin(machineDetailsView, machineDetailsView.machineId.equalsExp(machineView.id))
      ]);
  
}
```

**Comment by simolus3** [comment]
At the moment this is not possible.

I imagine this would also be hard to get right since references to view columns don't exist at the time the view is read. E.g. in your example, `machineView.providerId` would be a compile-time error when drift sees your code because the part file doesn't exist at that point. Tables don't have that problem since you define the columns beforehand, but some views have columns just because they were listed in the `select` clause. So in the end it's really hard to resolve columns between Dart views.
